### DISCUSSION TITLE
How to get the payload of another state within one input control

### DISCUSSION THREAD

**Discussion by saqwild** [writer/writer-framework#169] [Q&A]
Hello,
Understanding from the "Handling Input" documentation, 

```python
# This event handler takes the payload and assigns it
# to the state element "name"
def handle_input_change(state, payload):
    state["name"] = payload
``` 
will get the payload of the current control, however, at times, I need to refer to the payload of another state within the same (input) control.
Given the above example, I would need something like;

```python
# This event handler takes the payload and assigns it
# to the state element "name"
def handle_update(state, payload):
FirstName = payload
    state["name"] = FirstName
#trying to get the payload of last name
LastName = state["Last_Name"]
if LastName:
    Payload = payload
   state["Last_Name"] = payload
``` 
Thank you in advance

**Comment by FabienArcellier** [comment]
Hello,

I have investigated the code. I don't think there is a way to forward the target_component from `streamsync/core.py` inside the handler. The only way is to create one handler per component.

It may be interested to give access to the name and id of target component as context data. Currently it's not present.

**Reply by saqwild** [reply]
Hello Fabien,
Thank you for your time investigating, appreciated.
Even I entailed the same but wanted to get sort of confirmation on this.
I am not sure if this can be integrated easily and when but I am working on an application which heavily relies on other manual inputs, a plotly chart for instance.

```python
def t1_filter_hist(px, db_bdMaster, filteryaxis, filterxaxis):
    fig1 = px.histogram(
        db_bdMaster,
        x="Years",
        y=filteryaxis,
        histfunc=filterxaxis,
        text_auto=True,
        template="presentation",
        color_discrete_sequence=px.colors.qualitative.G10,
        marginal="box",
    )
    fig1.update_xaxes(type="category")
    fig1.update_yaxes(ticklabelposition="inside top", automargin=False)
    # fig1.update_traces(texttemplate="%{y:,.0f}")
    fig1.update_layout(
        autosize=True,
        xaxis=(dict(showgrid=True)),
        showlegend=False,
        font=dict(family="Courier New, monospace", size=14, color="Black"),
    )
    return fig1
``` 
as you will notice, y filter is another input (can be Actual or Budget) histfunction comes from another, (avg or sum) and even the dataframe.

do you know if there is a plan to integrate such option in future?

Thank you in advance
